simdata/simulation_analysis.ipynb

```diff
@@ -0,0 +1,754 @@
O06/fsP4tlne/PEE0/y4osDmTRpGkFBwYwb9wZTp07mmWf6cfr0aTp16kLt2nX54YflbN++lenTv8TpdPL0009wzz33AlCkSACffjqTn39ezdSpkxW4ItmwFCqK4/Z2JK+ZjnPfevzKN8AR2Znzc4aT/MuXFLq7t7dLLFAuzeLcymFQFouIXI1Lc9h0pgBg2Oz/et/KYRG5VjodwwO2bNlMo0aNCQpKv7BSmzYPsnnzhoznq1W7LWO9u+++F7vdTuHChZk2bRYlStwAQOPGdwJQpkxZzpw57dkDEMmH/CrfmX6Lzg1fY7rSsAbfhL1GC5x/rsN59HdvlydeoCwWEfEu5bCIgI/PhPCr2OiyzqzNZsHp9Oxt91yuS7/xM0lLS8tYcjj8AbDZbBgXXY7/2LGjBAen35bKarUCufsNoogvMywW/KOewLBYMSzpvz/2Wq1J/fMXktdMx/rwSAyrT0dgnnFpFnsjh0FZLCIF16U5nHb6GIZhYAkK92gdymERATfPhFi0aBEtWrTgvvvuY+bMmZc9v3z5ctq2bUubNm14+umnOXPmjDvL8Tir1UpaWhq1atVhzZpVxMenH9/ChfOpVavuZevXqFGblSt/wOl0kpSUxHPPPUtsbIynyxbxGdbiN2MJDsdMc+I6dwrDZse/0eO4Th8jZdt32W5b2/Yntax7PVSpuJOyWETEu5TDInIxt30NePz4ccaPH8+8efOw2+107NiR+vXrU758eSD9Vj0jRoxg7ty5hIWFMXHiRN59912GDRvmrpI8rnjxEoSFhTNx4lt07vwf+vTphdPppFKlKgwa9MJl60dF3cXu3bvo3r0TLpdJ+/aPcsstpb1QuYhvSVz8FmZaKoXbDsV2Sw1st9Yh5deF+JVrgKVoyBW3qWtTA8JXKItFRLxLOSwiFzNMN81j+uabb9i4cSOjR48G4P3338c0Tfr06QPAqVOn2LRpE/fem36RmSVLlrBo0SLef//9q36NkyfPXjatKzr6EOHhWYeUt6YBS+7I6ed7vUJCAomNTcj1/UrekLrnZ5JWTsbR5D/YK0fhOnuSc7NfxHpTZQo1649x8ZzP/9n64RAAajz1+jW9lsViUKJEQK7UnV9cmsXKYd+mHBZPOr9oDH5+NvzuH3TV2yiH02X3u+qt0zEk97gji6/n90183/WOi+yy2G2nY8TExBAS8s83jKGhoRw/fjxjuVixYhkNiKSkJCZNmkTTpk3dVY6IFGC2Cg2x3liJ5PWzcSUlYAkogaPOA6T9tRXnoV+vuE1gYT+CAv79VcNFREREROQfbjsdw+VyZfp20TTNK37bmJCQwDPPPEPlypV58MEHr+k1rtRZiYmxYLNl31vJ6XnJuywWCyEhgW7Zt7v2K3lDSusn+fuTgVi2ziek1dOYdz3E3/vXkfrLF4TXqI/FXijT+qlhRQGNCxERERGR3OS2JkR4eDibNm3KWI6NjSU0NDTTOjExMfTo0YMGDRrw4osvXvNrXGnqmcvlynaar6YB528ul8st03U1DbggKIa9ejMSti4mrfydWEuUwnZHZxIXjuLI0pn4N+iQae3UVCd+frZrHhcFcRqwiIiIiMjVctuUgIYNG7Ju3Tri4uJITExk2bJlNGnSJOP5tLQ0nnzySZo3b87QoUOvOEtCRCQ32Wu3wb/pM1iKlwTAFl4Bv0qNSd2+jLS4v71cnYiIiIiI73PbTIiwsDAGDBhAly5dSE1NpV27dkRERNCzZ0/69u1LdHQ0u3btIi0tjaVLlwJw2223MWrUKHeVJCIFnOHnj1/ZegC4zp/GUjgYe/1HSD34K8lrplOo9RAMQ6driYiIiIi4i9uaEACtW7emdevWmR6bPHkyANWrV2f37t3ufHkRkStK2b6U5E3fUOSRMViKFMNR/xGSV03Fuedn/Co19nZ5IiIiIiI+S1/55TGjR79CdPSxXN1nZGTdXN3fv9WuXWuOHTvK0aNHGDPmVW+XIwWQrXQtcKWRvO4LAPwqNcYSVp7kX77CTDrr5erE25TDIpKTRYsW0aJFC+677z5mzpx52fM7d+7k4Ycfpk2bNvTu3Zv4+HgA9u3bR6dOnWjbti0dOnTg999/B+DIkSPUqlWLtm3b0rZtW3r06OHR48mLlMUivktNiDzm1183YZpmziv6gOjoYxw5ovPwxfMsRUOx12yFc/8GnH/vwDAs+Ed2xUw5T/KGrwFwnT9D2tnTXq5UvEE5LCLZOX78OOPHj2fWrFnMnz+fr776ij///DPTOqNGjaJv374sXLiQMmXKMGXKFACGDRtGz549WbBgAf3792fw4MEA7Nixg9atW7NgwQIWLFiQsX5BpiwW8V1uPR1D0gN0xoyp+Pv7c/DgAcqVK8/LL49i+fKlfP31F7hcJpUqVea//x3M7NlfcOJELIMG9aNly7acOhXH00/3ZcOGXxg2bDCLF6/AZrPRqVM73n33Y44ePcrEiW+RkpJCcHAwgwa9SMmSpejTpxdFiwZx4MA+Xn11TEYt27dvZdSoEbz11juULFnqivUuXryItWtXc+rUKU6ePEGjRo3p02cAhmEwY8Y0fvzxe9LSXNSv34CnnupLdPQxXnxxIGXLlmPPnj8oXrwEI0e+TtGiQcyd+xVLliwmKSkRPz8/RowYxS233JrxWhMnvsXRo0cYN+4Nzp8/S40atWnTJv02rX369OKpp/pSrdptbv35SMFlr9Gc1L1rSfp5BkXavYa1RCn8bruX1O1L8avUGDPxDE5dMNcnKIeVwyK5ae3atTRo0IDg4GAAmjVrxpIlS+jTp0/GOi6Xi3PnzgGQmJhIUFAQAO3bt6dx4/TT/ipVqsSxY+nf9G/fvp09e/bQtm1bgoKCGDp0KJUqVfLkYbmdslhZLHKBmhAesGPHNmbOnMMNN4TQu3c3vvlmDitXruDDDz/F4XDw0Ufv8cUXM+jW7QkWLJjLm29OJCUlhVdfHQ7A5s0bcTgc7Nmzm+DgYhQpEkBgYFFGjOjGy
```

Comment:
Yay! Do you know why the pytorch implementations seem to have a higher variation in model performance than those of sklearn when there are many features?

Reply:
I'm not sure. The only difference that comes to mind is that I'm doing minibatch GD in pytorch and (I think) sklearn uses single-sample SGD. Not sure why that would cause more variability in pytorch though, I'd actually expect the opposite.

I'll have to think about this more - created #19 to remind myself.